src/main/clojure/clara/rules/logger.cljc

```diff
@@ -0,0 +1,25 @@
+(ns clara.rules.logger)
+
+(defn- warn-unused-binding
+  [{:keys [lhs rhs] :as production}]
+  (let [re-binding #"\?[\w-]+"
```

Comment:
I'd suggest using the [is-variable?](https://github.com/cerner/clara-rules/blob/0.20.0/src/main/clojure/clara/rules/compiler.clj#L82) function in the interest of consistency; we could factor it out into say an "analysis-utils" namespace to avoid a cyclic dependency.

Reply:
In the example below, `?cold` should get a warning for not being used. But in the production structure, the `:fact-binding` is a `keyword` therefore the definition of `is-variable?` does not work for it. I just verified with some other approach, but it should be considered a variable or not? [thus `is-variable` needs to be extended to handle not only symbols but also keywords]

```clj
(rl/defrule rule-testing
    [?cold <- ColdAndWindy (= ?w windspeed) (< temperature 10)
                           (= ?w 20)]
    =>
    (println "All bindings used"))
```